### PR TITLE
fix(website): mobile navigation buttons in dark mode

### PR DESCRIPTION
**Pull request by shleewhite**
Currently, if you are in dark mode on mobile, the hamburger/close icons for the navigation are too low contrast:
Screen Shot 2022-04-05 at 10 07 57 PM
Screen Shot 2022-04-05 at 10 08 01 PM

This PR:
* rewrites the Pattie component to use styled components so that I can use icon color tokens as the background color
* adds stories for the toggled/not toggle hamburger

The button in the header uses `colorTextIconBrandInverse`
Light mode:
Screen Shot 2022-04-05 at 10 03 35 PM
Dark mode:
Screen Shot 2022-04-05 at 10 03 58 PM

The button in the navigation popover uses `colorTextIcon`
Light mode:
Screen Shot 2022-04-05 at 10 03 40 PM
Dark mode:
Screen Shot 2022-04-05 at 10 04 03 PM

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: packages/paste-website/src/components/site-wrapper/site-header/HamburgerToggle.tsx [hunk 1]
```diff
@@ -1,53 +1,60 @@
...
-    transition="all .25s ease-in-out"
-    transformOrigin="left center"
-    {...props}
-  />
-);
+interface StyledPattieProps extends Pick<BoxProps, 'top' | 'left' | 'transform' | 'opacity'> {
+  color: TextColorOptions;
+}
+
+const StyledPattie = styled.span<StyledPattieProps>`
```
Comment (TheSisb) [inline packages/paste-website/src/components/site-wrapper/site-header/HamburgerToggle.tsx:16]: suggestion: we've been moving away from themeGet to the styled+css combo for performance and ease of development. Typing out `${themeGet()}` all over gets messy heh

```js
const StyledPattie = styled.span<StyledPattieProps>(({color, ...props}) =>
  css({
    position: 'absolute',
    width: 'space.space50',
    height: '3px',
    borderRadius: 'radii.borderRadius30',
    transition: 'transform 0.25s ease-in-out, opacity 0.25s ease-in-out',
    transformOrigin: 'left center',
    backgroundColor: `textColors.${color}`,
    ...props
  })
);
```

File: packages/paste-website/stories/HamburgerToggle.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+import type {StoryFn} from '@storybook/react';
+import {HamburgerToggle} from '../src/components/site-wrapper/site-header/HamburgerToggle';
+
+export default {
+  title: 'Website/HamburgerToggle',
+};
+
+export const ToggledHamburger: StoryFn = () => <HamburgerToggle toggled color="colorTextIcon" />;
```
Comment (TheSisb) [inline packages/paste-website/stories/HamburgerToggle.stories.tsx:9]: praise: Good addition!

**Review by TheSisb — APPROVED** [review]

File: packages/paste-website/src/components/site-wrapper/site-header/HamburgerToggle.tsx (hunk 1 above)
Comment (shleewhite) [reply packages/paste-website/src/components/site-wrapper/site-header/HamburgerToggle.tsx:16]: I initially tried doing it this way, but I think the CSS function doesn't let you use text colors for background color. I just tried it again and It doesn't set the background color at all:
Screen Shot 2022-04-06 at 11 11 19 AM

Comment (SiTaggart) [reply packages/paste-website/src/components/site-wrapper/site-header/HamburgerToggle.tsx:16]: Yeah, it won't. The CSS function is keyed to the theme and style props. So you can only use backgrounds for background and colors for colors. Mainly, so you don't do something that isn't intended by the design language. 

Here it's fine, as you are building an icon from backgrounds. It's a funky edge case

**Review by SiTaggart — APPROVED** [review]

**Review by loreina — APPROVED** [review]

File: packages/paste-website/src/components/site-wrapper/site-header/HamburgerToggle.tsx (hunk 1 above)
Comment (TheSisb) [reply packages/paste-website/src/components/site-wrapper/site-header/HamburgerToggle.tsx:16]: I see, I hadn't noticed that.

**Pull request merged**